### PR TITLE
feat: Alerts refactor - create new Alert component for admin/FormBuilder

### PR DESCRIPTION
**Pull request by dsamojlenko**
# Summary | Résumé

Create a new Alert component for use on the Admin/FormBuilder side of the app. To replace the current `globals/Alert.tsx` and `globals/Attention/Attention.tsx` components.

Includes extensive set of Cypress Component tests that serve as a good reference for how to use the Alert component.

Intention is to streamline all Admin/FormBuilder Alerts to line up with the figma designs and create a flexible and customizable component that can be used for each of the different alert styles unique to the Admin/FormBuilder section.

### Simple Alert - includes default icon (see tests)
```
<Alert.Success title="This is a title" body="This is a body" />
<Alert.Warning title="This is a title" body="This is a body" />
<Alert.Info title="This is a title" body="This is a body" />
<Alert.Danger title="This is a title" body="This is a body" />
```
image

### Custom icon
```
<Alert.Success
    title="This is a title"
    body="This is a body"
    icon={<CircleCheckIcon className="mr-1 h-12 w-12" />}
/>
```

### No icon
```
<Alert.Warning title="This is a title" body="This is a body" icon={false} />
```
image

### Dismissible - includes onDismiss action that removes the alert
```
<Alert.Success dismissible title="This is a title" body="This is a body" />
```

image

### Dismissible - custom onDismiss action
```
<Alert.Success
    dismissible
    onDismiss={onDismissHandler}
    title="This is a title"
    body="This is a body"
/>
```

### Focussable
```
<Alert.Success focussable title="This is a title" body="This is a body" />
```

### Complex alerts
Alerts are highly customizable by providing children components for Icon, Title, and Body
```
<Alert.Warning title="This is a title" body="This is a body">
    <Alert.IconWrapper>
        <CircleCheckIcon />
    </Alert.IconWrapper>
    <Alert.Title>Test Title</Alert.Title>
    <Alert.Body>Test body</Alert.Body>
</Alert.Warning>
```

Each of these inner components can accept custom class overrides, and you can specify a heading level on the Title
```
<Alert.Success className="mt-8>
    <Alert.Title headingTag="h4" className="pb-4 text-xl">
        Test Title
    </Alert.Title>
    <Alert.Body className="p-4">Test body</Alert.Body>
</Alert.Success>
```

The body of the Alert can just be any arbitrary elements, but not plain text
```
<Alert.Success title="This is an Alert">
    <p>With a paragraph</p>
    <ul>
        <li>And a list</li>
        <li>Another list item</li>
    </ul>
    <>Here is some text wrapped in a fragment</>
    <p>
        This paragraph <a href="/nowhere">contains a link</a>.
    </p>
    <Button className="mt-8">And a button</Button>

    This text will not render
</Alert.Success>
```
image

## Design
https://www.figma.com/file/93udAjdrHbzOOCxaC5lqN1/GC-Forms-component-library?type=design&node-id=19-640&mode=design&t=pC0kVSSf7AG1PKqi-0

## cn()
Includes a new `cn` util function for resolving Tailwindcss priority/conflicts. When passing in custom classes to the various components, instead of just appending them to the end of existing classes, this utility will merge them and ensure classes added later will take precedence.

## TODO
This PR just includes the new component and related code. Future PRs will implement this Alert component everywhere that `components/Alert.tsx` and `components/Attention/Attention.tsx` are currently being used.

### CODE REVIEW

File: components/globals/Alert/Alert.tsx [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+  INFO = "INFO",
+}
+
+export const Title = ({
+  children,
+  level = "h2",
+  status,
+}: {
+  children: string;
+  level?: "h2" | "h3" | "h4";
```
Comment (timarney) [inline components/globals/Alert/Alert.tsx:18]: Should be able to do what we do in ErrorPanel here where we alias to an uppercase prop

```
 headingTag: HeadingTag = "h2",
 ```
 
 and than we can use 
 
 ```
  <HeadingTag className="border-none mb-4">{title || defaultTitle}</HeadingTag>
 ```
 
 vs the current
 
 ```
  const Wrapper = `${level}` as keyof JSX.IntrinsicElements;
 ```

File: components/globals/Alert/Alert.tsx [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+    return <>{children}</>;
+  }
+  const Wrapper = `${level}` as keyof JSX.IntrinsicElements;
+  const className = status ? `${defaultClasses.text[status]}` : "";
+
+  return <Wrapper className={`${className}`}>{children}</Wrapper>;
+};
+
+export const Body = ({ children }: { children: JSX.Element | string }) => {
+  return <>{children}</>;
```
Comment (timarney) [inline components/globals/Alert/Alert.tsx:31]: Do we need the React fragment here?  I don't see a case where we could pass children without 

```
<Body>
            <>
              {alertBody && <>{alertBody}</>}
              {content}
            </>
          </Body>
```

File: components/globals/Alert/Alert.tsx [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+
+export const Icon = ({
+  children,
+  status,
+}: {
+  children: JSX.Element;
+  status?: ErrorStatus | undefined;
+}) => {
+  if (children.type.name === "Icon") {
+    return <>{children}</>;
```
Comment (timarney) [inline components/globals/Alert/Alert.tsx:42]: Same fragment comment as above

File: components/globals/Alert/Alert.tsx [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+    [ErrorStatus.WARNING]: "text-yellow-700",
+    [ErrorStatus.ERROR]: "text-red-700",
+    [ErrorStatus.INFO]: "text-indigo-700",
+  },
+};
+
+const defaultIcons = {
+  [ErrorStatus.SUCCESS]: (
+    <Icon status={ErrorStatus.SUCCESS}>
+      <CircleCheckIcon className={`h-12 w-12`} />
```
Comment (timarney) [inline components/globals/Alert/Alert.tsx:73]: wondering if we should add a const or something for the className so each case can just use vs "className={`h-12 w-12`} />" on each

File: components/globals/Alert/Alert.tsx (hunk 1 above)
Comment (dsamojlenko) [reply components/globals/Alert/Alert.tsx:18]: that's cool

**Review by timarney — APPROVED** [review]
Nice work @dsamojlenko 

Let's get this merged and than implement a first one here as per the design in the security questions branch.

https://www.figma.com/file/89fRce4gmy64UIArSZoAM8/v3.-GC-Forms-(builder-and-vault)?type=design&node-id=1783-78150&mode=design&t=kypKK3BIN3a4xKfw-0

**Pull request merged**